src/components/ProductFilterModal.vue

```diff
@@ -84,26 +92,43 @@ export default defineComponent({
       modalController.dismiss({ dismissed: true, isFilterChanged: this.isFilterChanged });
     },
     async search(event: any) {
-      // TODO: need to implement infinite scroll on the modal search
+      this.queryString = event.target.value;
+      this.getTags();
+    },
+    async getTags(vSize?: any, vIndex?: any) {
+      const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
+      const viewIndex = vIndex ? vIndex : 0;
+
       const payload = {
         facetToSelect: this.facetToSelect,
         docType: 'PRODUCT',
         coreName: 'enterpriseSearch',
         searchfield: this.searchfield,
         jsonQuery: '{"query":"*:*","filter":["docType:PRODUCT"]}',
         noConditionFind: 'N',
-        limit: 50,
-        q: event.target.value,
-        term: event.target.value
+        limit: viewSize,
+        q: this.queryString,
+        term: this.queryString,
+        offset: viewIndex * viewSize,
       }
 
       const resp = await ProductService.fetchFacets(payload);
       if (resp.status == 200 && resp.data.length > 0) {
-        this.facetOptions = resp.data.map((obj: any) => ({ id: obj.id, label: obj.label }))
+        this.facetOptions = viewIndex === 0 ? resp.data : [...this.facetOptions , ...resp.data];
+        this.isScrollable = (this.facetOptions.length % process.env.VUE_APP_VIEW_SIZE) === 0 ? true : false;
```

Comment:
I think this will do the same
```suggestion
        this.isScrollable = this.facetOptions.length % process.env.VUE_APP_VIEW_SIZE) === 0;
```